test/puppetlabs/puppetdb/http/command_test.clj

```diff
@@ -139,3 +136,110 @@
         (let [timestamp (get-in good-msg [:headers :received])]
           (time/parse (time/formatters :date-time) timestamp))))))
 
+(deftest normalize-request
+  (let [normalize (#'tgt/normalize-request identity)]
+
+    ;; Check for an error if some but not all params are present
+    (doseq [items (apply concat
+                         (map #(combinations ["command" "version" "certname"] %)
+                              [1 2]))]
+      (let [result (normalize {:body ::ignored
+                               :params (into {} (for [k items] [k "1"]))})]
+        (is (= 400 (:status result)))
+        (is (= (apply str
```

Comment:
Hmm, well I wanted the test to make sure that the string actually matches the error case.  I suppose I could have chopped off the bit after the colon and turned that into a set for comparison.  Would that be preferable?